examples/material_gallery/test/smoke_test.dart

```diff
@@ -8,55 +8,122 @@ import 'package:test/test.dart';
+// called 'name' is already open.
+void smokeDemo(WidgetTester tester, String menuItemText) {
+  // Ensure that we're (likely to be) on the home page
+  final Finder navigationMenuButton = findNavigationMenuButton(tester);
+  expect(tester, hasWidget(navigationMenuButton));
+
+  tester.tap(find.text(menuItemText));
+  tester.pump(); // Launch the demo.
+  tester.pump(const Duration(seconds: 1)); // Wait until the demo has opened.
+
+  // Go back
+  Finder backButton = findBackButton(tester);
+  expect(tester, hasWidget(backButton));
+  tester.tap(backButton);
+  tester.pump(); // Start the navigator pop "back" operation.
+  tester.pump(const Duration(seconds: 1)); // Wait until it has finished.
+}
+
 void main() {
   test('Material Gallery app smoke test', () {
     testWidgets((WidgetTester tester) {
       material_gallery.main(); // builds the app and schedules a frame but doesn't trigger one
       tester.pump(); // see https://github.com/flutter/flutter/issues/1865
       tester.pump(); // triggers a frame
 
-      // Try loading Weather demo
-      tester.tap(find.text('Demos'));
-      tester.pump();
-      tester.pump(const Duration(seconds: 1)); // wait til it's really opened
+      // Expand the demo category submenus.
+      for(String category in demoCategories.reversed) {
+        tester.tap(find.text(category));
+        tester.pump();
+        tester.pump(const Duration(seconds: 1)); // Wait until the menu has expanded.
+      }
 
-      tester.tap(find.text('Weather'));
-      tester.pump();
-      tester.pump(const Duration(seconds: 1)); // wait til it's really opened
-
-      // Go back
-      Finder backButton = find.byElement((Element element) {
-        Widget widget = element.widget;
-        if (widget is Tooltip)
-          return widget.message == 'Back';
-        return false;
-      });
-      expect(tester, hasWidget(backButton));
-      tester.tap(backButton);
-      tester.pump(); // start going back
-      tester.pump(const Duration(seconds: 1)); // wait til it's finished
-
-      // Open menu
-      Finder navigationMenu = find.byElement((Element element) {
-        Widget widget = element.widget;
-        if (widget is Tooltip)
-          return widget.message == 'Open navigation menu';
-        return false;
-      });
-      expect(tester, hasWidget(navigationMenu));
-      tester.tap(navigationMenu);
-      tester.pump(); // start opening menu
-      tester.pump(const Duration(seconds: 1)); // wait til it's really opened
+      final List<double> scrollDeltas = new List<double>();
+      double previousY = tester.getTopRight(find.text(demoCategories[0])).y;
+      for(String name in demoNames) {
+        final double y = tester.getTopRight(find.text(name)).y;
+        scrollDeltas.add(previousY - y);
+        previousY = y;
+      }
+
+      // Launch each demo and then scroll that item out of the way.
+      for(int i = 0; i < demoNames.length; i++) {
```

Comment:
`i += 1` (or `++i`, but I prefer the former). `i++` is a bad habit because in poor compilers it's more expensive (since the old value is first stored before the variable is incremented)